### ISSUE TITLE
step_counter not set

### ISSUE THREAD

**Issue by m-toman** [lessw2020/Ranger-Deep-Learning-Optimizer#8]
Hi,
thanks for your work.

I just plugged it into my model and found that step_counter was not set for all param_groups.

I fixed it with this hack:
```
        #look ahead tracking and updating if latest batch = k
        for group,slow_weights in zip(self.param_groups,self.slow_weights):
            if 'step_counter' not in group:
                group["step_counter"] = 0
```
but I suspect it's not optimal... 
this would mean that self.param_groups changed between the constructor and step(), but I have no idea why. Have you seen something similar before?

Thanks

**Comment by lessw2020** [comment]
Hi @m-toman,
I've rewritten the whole slow_weight buffer and handling now.  We leverage the already existing step counter in RAdam, and updates are handled at the param pass level rather than revisiting with a whole new loop.
This should correct the issue you noticed!

**Comment by lessw2020** [comment]
Closing, but please let me know if issue is not resolved with this latest version.

**Comment by m-toman** [comment]
Thanks, it works now.
I'm currently re-running an experiment for audio synthesis (with one of the WaveRNN derivates) and looks like this until now:

Adam:
![image](https://user-images.githubusercontent.com/9433378/64317860-1af9f900-cfb9-11e9-8fb8-84970ba9f0ce.png)

Ranger:
![image](https://user-images.githubusercontent.com/9433378/64317882-26e5bb00-cfb9-11e9-80a0-b70a5cd1643f.png)

So the loss value is pretty much the same at the current point but the trajectory is much smoother.
I got originally good results at about 1 million steps, so some time to go to really see the differences.

**Comment by entruv** [comment]
You are logging 1 loss every iterations for the last graph, and 1 loss
every ~1000 iterations for the first graph.

Is this the only change in your model ?
Is this the number of epochs in the x-axis ?

**Comment by m-toman** [comment]
Yeah I guess the problem is tensorboard picks some.. sampling rate for the x axis because the original model was trained for a couple million epochs (with an epoch being about 100 batches of size 128 audio sequence snippets), so it doesn't display them all. Doesn't help because it still regularly OOMs and sometimes takes the instance with it. 

Anyway, the optimizer should really be the only difference between the two models except for some randomness that might not have started with the same seed (data set shuffling and random snippets from each audio file).

And right, I picked the epoch plot because it's less noisy. So I meant epochs when I wrote steps.
